### PR TITLE
Add org.ioquake3.ioquake3

### PR DESCRIPTION
**Pull request by AsciiWolf**
### Please confirm your submission meets all the criteria

- [x] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [x] My pull request follows the instructions at [App Submission][submission].
- [x] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [x] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [x] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub. **Link:** https://github.com/ioquake/ioq3/issues/616
- [ ] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
- [x] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*

[reqs]: https://docs.flathub.org/docs/for-app-authors/requirements
[maint]: https://docs.flathub.org/docs/for-app-authors/maintanance
[submission]: https://docs.flathub.org/docs/for-app-authors/submission

I am submitting this Flatpak on behalf of @mattiasb who is the original author, but is too busy at the moment and [allowed](https://github.com/mattiasb/flathub/pull/2#issuecomment-1721277455) me to submit it.

### CODE REVIEW

**Comment by AsciiWolf** [comment]
bot, build org.ioquake3.ioquake3

File: org.ioquake3.ioquake3.yaml [hunk 1]
```diff
@@ -0,0 +1,57 @@
+app-id: org.ioquake3.ioquake3
+runtime: org.freedesktop.Platform
+runtime-version: '22.08'
```
Comment (hfiguiere) [inline org.ioquake3.ioquake3.yaml:3]:
```suggestion
runtime-version: '23.08'
```

File: org.ioquake3.ioquake3.yaml [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+  - name: ioquake3
+    buildsystem: simple
+    build-options:
+      env:
+        BUILD_SERVER: "0"
+        COPYDIR: /app/games/quake3
+      arch:
+        i386:
+          env:
+            IOQ3_ARCH: x86
```
Comment (hfiguiere) [inline org.ioquake3.ioquake3.yaml:25]:
```suggestion
```

this arch was dropped long before `arm`

File: org.ioquake3.ioquake3.yaml [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+  - name: ioquake3-launcher
+    buildsystem: simple
+    build-commands:
+      - install -Dm744 ioquake3 -t /app/bin
+    sources:
+      # ioquake3 launcher script that:
+      # - … makes ~/.q3a/ available at ~/.var/app/org.ioquake3.ioquake3/data/
+      # - … explains (via zenity) where the user can put configs and pk3-files.
+      - type: file
+        path: ioquake3
```
Comment (hfiguiere) [inline org.ioquake3.ioquake3.yaml:57]:
```suggestion
      - install -Dm644 misc/setup/org.ioquake3.ioquake3.metainfo.xml
                -t /app/share/metainfo
      - install -Dm744 ioquake3 -t /app/bin
    sources:
      - type: git
        url: https://github.com/ioquake/ioq3.git
        commit: b1e6ef14254fdcb77cbe2f602c22549d3fb0eb78
      - type: patch
        path: ioquake3_appstream.patch
      # ioquake3 launcher script that:
      # - … makes ~/.q3a/ available at ~/.var/app/org.ioquake3.ioquake3/data/
      # - … explains (via zenity) where the user can put configs and pk3-files.
      - type: file
        path: ioquake3
```

File: org.ioquake3.ioquake3.yaml (hunk 1 above)
Comment (AsciiWolf) [reply org.ioquake3.ioquake3.yaml:3]: Yep, I will do it, but I need to test it first. :-)

**Comment by mattiasb** [comment]
Thank you @AsciiWolf for both your work, your patience with me and for taking this over. We should have done so from the beginning. :)

**Comment by AsciiWolf** [comment]
bot, build org.ioquake3.ioquake3

**Comment by AsciiWolf** [comment]
> Build [66208 successful](https://buildbot.flathub.org/#/builders/6/builds/66208) To test this build, install it from the testing repository

Tested and works great. :-)

**Comment by AsciiWolf** [comment]
@barthalion Thanks for the quick review and merge! Please, also send an invitation to @mattiasb who is the original author of this Flatpak.

**Pull request closed without merging**